include/maidsafe/common/convert.h

```diff
@@ -0,0 +1,93 @@
+/*  Copyright 2012 MaidSafe.net limited
+
+    This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
+    version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+    licence you accepted on initial access to the Software (the "Licences").
+
+    By contributing code to the MaidSafe Software, or to this project generally, you agree to be
+    bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
+    directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
+    available at: http://www.maidsafe.net/licenses
+
+    Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
+    under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+    OF ANY KIND, either express or implied.
+
+    See the Licences for the specific language governing permissions and limitations relating to
+    use of the MaidSafe Software.                                                                 */
+
+#ifndef MAIDSAFE_COMMON_CONVERT_H_
+#define MAIDSAFE_COMMON_CONVERT_H_
+
+#include "asio/ip/address.hpp"
+#include "asio/ip/address_v4.hpp"
+#include "asio/ip/address_v6.hpp"
+#include "asio/ip/udp.hpp"
+
+#include "boost/asio/ip/address.hpp"
+#include "boost/asio/ip/address_v4.hpp"
+#include "boost/asio/ip/address_v6.hpp"
+#include "boost/asio/ip/udp.hpp"
+
+namespace maidsafe { namespace common { namespace convert {
+
+// Warning: Will map error from any error_category to an error
+// in the generic_category. Even boost asio category maps to
+// the std generic category so there probably isn't a bijection
+// between the std::error_code and the boost::system::error_code.
+inline std::error_code ToStd(const boost::system::error_code& ec) {
```

Comment:
I would have preferred the "fix when it becomes a problem" approach, but
changed it anyway.

On Mon, Feb 9, 2015 at 5:58 PM, Fraser Hutchison <notifications@github.com>
wrote:

> In include/maidsafe/common/convert.h
> <https://github.com/maidsafe/MaidSafe-Common/pull/117#discussion_r24343982>
> :
>
> > +#include "asio/ip/address_v4.hpp"
> > +#include "asio/ip/address_v6.hpp"
> > +#include "asio/ip/udp.hpp"
> > +
> > +#include "boost/asio/ip/address.hpp"
> > +#include "boost/asio/ip/address_v4.hpp"
> > +#include "boost/asio/ip/address_v6.hpp"
> > +#include "boost/asio/ip/udp.hpp"
> > +
> > +namespace maidsafe { namespace common { namespace convert {
> > +
> > +// Warning: Will map error from any error_category to an error
> > +// in the generic_category. Even boost asio category maps to
> > +// the std generic category so there probably isn't a bijection
> > +// between the std::error_code and the boost::system::error_code.
> > +inline std::error_code ToStd(const boost::system::error_code& ec) {
>
> Yes, but we generally aim to include what we use. Say
> boost/asio/ip/udp.hpp is the only file which brings
> boost::system::error_code into scope here and we either no longer include
> it, or it changes to no longer include boost/system/error_code.hpp. Then
> that unrelated change breaks this file.
>
> It's definitely more of an issue for headers rather than source files.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/maidsafe/MaidSafe-Common/pull/117/files#r24343982>.
>